test/dataset_test.cpp

```diff
@@ -464,7 +464,8 @@ TEST(Dataset, operator_times_equal_histogram_data) {
   b.insert(Data::Variance, "name1", {Dim::X, 1}, {4.0});
 
   // Counts (aka "histogram data") times counts not possible.
-  EXPECT_THROW_MSG(a *= a, std::runtime_error, "Unsupported unit on LHS");
+  EXPECT_THROW_MSG(a *= a, std::runtime_error,
```

Comment:
Testing looks incomplete here.

Reply:
@OwenArnold The goal here is to do a minimal refactor of the internals of `Unit`. Once that is done we will extend (and probably refactor, e.g., we are considering removing the Unit IDs in favor of constants) the client code.